elements/includes/update.batch.inc

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * @file
+ * Batch to update all forms.
+ */
+
+/**
+ * Batch operation to rename all datepicker elements to the new value.
+ *
+ * @param array $context
+ *   An array containing the context of the batch operation.
+ */
+function xml_form_elements_rename_datepicker_batch_operation(&$context) {
+  $sandbox = &$context['sandbox'];
+
+  if (!isset($sandbox['forms'])) {
+    $sandbox['forms'] = db_select('xml_forms', 'x')
+      ->fields('x')
+      ->execute()
+      ->fetchAllAssoc('id', PDO::FETCH_ASSOC);
+    if (empty($sandbox['forms'])) {
+      return;
+    }
+  }
+  // Process a form at a time.
+  $form = array_shift($sandbox['forms']);
+  $context['message'] = t('Parsing @form_name for datepicker elements.', array(
+    '@form_name' => $form['name'],
+  ));
+  $dom = new DOMDocument();
+  $dom->loadXML($form['form']);
+
+  $xpath = new DOMXPath($dom);
+  $results = $xpath->query("//type[text() = 'datepicker']");
```

Comment:
Feels like the XPath here may be a little too broad? `//element/properties/type[text()='datepicker']` may be safer?